distributionviewer/core/static/js/app/components/containers/chart-detail-container.js

```diff
@@ -16,20 +17,25 @@ class ChartDetailContainer extends React.Component {
   }
 
   render() {
-    return (
-      <ChartDetail
-        id={this.chartId}
-        metric={this.props.metric.metric}
-        points={this.props.metric.points}
-        type={this.props.metric.type}
-      />
-    );
+    if (this.props.status === 404) {
+      return <NotFound />;
+    } else {
+      return (
+        <ChartDetail
+          id={this.chartId}
+          metric={this.props.metric.metric}
+          points={this.props.metric.points}
+          type={this.props.metric.type}
+        />
+      );
+    }
   }
 }
 
 const mapStateToProps = function(store) {
   return {
-    metric: store.metricState.metric
+    metric: store.metricState.metric,
+    status: store.metricState.status,
```

Comment:
I wasn't expecting such a detailed reply but thanks. Interesting to know and see the babel conversion live. That's cool.